### PR TITLE
lock rootfs on startup

### PR DESCRIPTION
**Pull request by ArsenArsen**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -221,6 +223,15 @@ fn run_init(cfg: &Config) -> ! {
 fn main() {
     let cfg = make_config_from_cli();
 
+    let root_dir = open(
+        &cfg.rootfs,
+        OFlag::O_RDONLY | OFlag::O_DIRECTORY | OFlag::O_CLOEXEC,
+        Mode::from_bits(0o777).expect("what?"),
```
Comment (avdgrinten) [inline src/main.rs:229]: `.expect("what?")` can be replaced by `.unwrap()` ;D

Comment (ArsenArsen) [reply src/main.rs]: oh I didn't remove that

**Comment by ArsenArsen** [comment]
another thing for consideration would be to use a file in the root as opposed to the root itself for this? since the following paradigm leads to an issue:

```sh
exec 4>/some/root/directory  # /some/root/directory: Is a directory
flock 4
```

but would that require extra permissions to the rootfs? is this better than various hacks for using `flock` as a wrapper (as `flock` as a wrapper handles directories correctly).

another consideration is should this file be CLOEXEC'd? can some process exist after `cbuildrt`s process die, hence leading to the lock being released too early?

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Pull request merged**